### PR TITLE
openstack-ardana: add support for CentOS 7.5 compute

### PR DESCRIPTION
**Pull request by flaviodsr**
Implements SCRD-3017

This change adds support for deploying ardana with CentOS 7.5 computes
on both virtual and bare metal deployments. In addition to that, the change
also adds a new parameter to openstack-ardana jobs and backend support
that can be used to select the OS distribution used for RHEL compute nodes.

Preliminary support for RHEL 7.5 is also included, although it hasn't been
tested yet because it requires licences and support for CentOS 7.3 is preserved
but deprecated. 

For bare metal the following image is used to reimage the nodes:
  - http://10.84.144.252/compute_iso/CentOS-75-x86_64-DVD-1804.iso

For virtual a new centos75 image was created on the ECP. The following
changes was maded to the image:

  1. Image: http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1804_02.qcow2
  2. Enabled root ssh with password
  3. Disabled CentOS repositories (Base and Updates)
  4. Removed hard coded mac address: https://docs.openstack.org/image-guide/openstack-images.html#no-hard-coded-mac-address-information
  5. Removed the following packages:
     - PyYAML
     - python-prettytable
     - python-backports-ssl_match_hostname
  6. Enabled SELinux
  7. Relabeled the file system: https://www.centos.org/docs/5/html/5.2/Deployment_Guide/sec-sel-fsrelabel.html

### CODE REVIEW

File: scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/defaults/main.yml [hunk 1]
```diff
@@ -21,13 +21,13 @@ sles_media_server: "ibs-mirror.prv.suse.net"
...
-    download_when: "{{ cloud_release == 'cloud8' }}"
+    download_when: "{{ cloud_release == 'cloud8' and is_physical_deploy }}"
   - name: "sles12sp4.iso"
     url: "http://{{ sles_media_server }}/install/SLE-12-SP4-Server-RC1/SLE-12-SP4-Server-DVD-x86_64-RC1-DVD1.iso"
-    download_when: "{{ cloud_release == 'cloud9' }}"
+    download_when: "{{ cloud_release == 'cloud9' and is_physical_deploy }}"
   - name: "rhel7.iso"
-    url: "http://10.84.144.252/compute_iso/RHEL-7.3-20170728-Server-x86_64-dvd.iso"
+    url: "http://10.84.144.252/compute_iso/CentOS-75-x86_64-DVD-1804.iso"
     download_when: "{{ rhel_enabled }}"
```
Comment (stefannica) [inline scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/defaults/main.yml]: `and is_physical_deploy` ?

File: scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/defaults/main.yml [hunk 2]
```diff
@@ -21,13 +21,13 @@ sles_media_server: "ibs-mirror.prv.suse.net"
...
   - name: "rhel7-opt-yum.tgz"
-    url: "http://ardana.suse.provo.cloud/yum-extras/yum-internal-rhel73-20180118-1320.tgz"
+    url: "http://ardana.suse.provo.cloud/yum-extras/yum-internal-rhel75-20181026-1528.tgz"
```
Comment (stefannica) [inline scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/defaults/main.yml]: same here ?

File: scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/tasks/main.yml [hunk 3]
```diff
@@ -39,7 +39,6 @@
     remote_src: yes
 
 - include_tasks: download_cobbler_requires.yml
```
Comment (stefannica) [inline scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/tasks/main.yml]: Oh, I see what you did there. So why do we need the CentOS DVD on the deployer node for non-cobbler deployments ?

Comment (flaviodsr) [reply scripts/jenkins/ardana/ansible/roles/bootstrap_ardana/tasks/main.yml]: The CentOS DVD is mounted on the deployer and served as a repo to the CentOS computes as in [1].

1. https://github.com/SUSE-Cloud/automation/pull/2956/files#diff-882ace793baa149c2b8cad9c43889d6bR160

**Pull request merged**